src/Config/RuntimeConfigLoader.cs

```diff
@@ -235,4 +241,52 @@ internal static string GetConnectionStringWithApplicationName(string connectionS
         // Return the updated connection string.
         return connectionStringBuilder.ConnectionString;
     }
+
+    /// <summary>
+    /// It adds or replaces a property in the connection string with `Application Name` property.
+    /// If the connection string already contains the property, it appends the property `Application Name` to the connection string,
+    /// else add the Application Name property with DataApiBuilder Application Name based on hosted/oss platform.
+    /// </summary>
+    /// <param name="connectionString">Connection string for connecting to database.</param>
+    /// <returns>Updated connection string with `Application Name` property.</returns>
+    internal static string GetPgSqlConnectionStringWithApplicationName(string connectionString)
+    {
+        // If the connection string is null, empty, or whitespace, return it as is.
+        if (string.IsNullOrWhiteSpace(connectionString))
+        {
+            return connectionString;
+        }
+
+        string applicationName = ProductInfo.GetDataApiBuilderUserAgent();
+
+        // Create a StringBuilder from the connection string.
+        NpgsqlConnectionStringBuilder connectionStringBuilder;
+        try
+        {
+            connectionStringBuilder = new NpgsqlConnectionStringBuilder(connectionString);
+        }
+        catch (Exception ex)
+        {
+            throw new DataApiBuilderException(
+                message: DataApiBuilderException.CONNECTION_STRING_ERROR_MESSAGE,
+                statusCode: HttpStatusCode.ServiceUnavailable,
+                subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization,
+                innerException: ex);
+        }
+
+        // If the connection string does not contain the `Application Name` property, add it.
+        // or if the connection string contains the `Application Name` property, replace it with the DataApiBuilder Application Name.
+        if (connectionStringBuilder.ApplicationName.IsNullOrEmpty())
```

Comment:
Looks like compared to `SqlConnectionStringBuilder`, `NpgsqlConnectionStringBuilder` doesn't set a default value for ApplicationName. Good catch!